### PR TITLE
HtmlEditor - support context menu customization

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

**Review by RomanTsukanov — DISMISSED** [review]

File: js/ui/html_editor/modules/tableContextMenu.js [hunk 1]
```diff
@@ -42,62 +52,94 @@ if(Quill) {
...
-                            { text: localizationMessage.format('dxHtmlEditor-deleteRow'), icon: 'deleterow', onClick: getTableOperationHandler(this.quill, 'deleteRow') },
-                            { text: localizationMessage.format('dxHtmlEditor-deleteTable'), icon: 'deletetable', onClick: getTableOperationHandler(this.quill, 'deleteTable') }
-                        ]
-                    },
-                    { text: localizationMessage.format('dxHtmlEditor-cellProperties'), icon: 'cellproperties', onClick: (e) => { this.showCellProperties(e); } },
-                    { text: localizationMessage.format('dxHtmlEditor-tableProperties'), icon: 'tableproperties', onClick: (e) => { this.showTableProperties(e); } }
-                ]
+                text: localize(buttonText),
+                icon: iconName.toLowerCase(),
+                onClick: this._formatHandlers[name] || getDefaultClickHandler(name, this)
```
Comment (DokaRus) [inline js/ui/html_editor/modules/tableContextMenu.js]:
```suggestion
                onClick: this._formatHandlers[name] ?? getDefaultClickHandler(name, this)
```

File: js/ui/html_editor/modules/tableContextMenu.js [hunk 2]
```diff
@@ -42,62 +52,94 @@ if(Quill) {
...
-                this.quill.focus();
-                return table[operationName](...rest);
+            });
+
+            return resultItems;
+        }
+
+        _getMenuConfig(items) {
+            const defaultItems = [
+                { text: 'Insert', items: [
```
Comment (DokaRus) [inline js/ui/html_editor/modules/tableContextMenu.js]: >text: 'Insert'

Localization?

File: js/ui/html_editor/modules/tableContextMenu.js [hunk 3]
```diff
@@ -42,62 +52,94 @@ if(Quill) {
...
+                    'insertHeaderRow',
+                    'insertRowAbove',
+                    'insertRowBelow',
+                    extend(this._prepareMenuItemConfig('insertColumnLeft'), { beginGroup: true }),
+                    'insertColumnRight'
+                ] },
+                {
+                    text: 'Delete',
```
Comment (DokaRus) [inline js/ui/html_editor/modules/tableContextMenu.js]: >text: 'Delete',

Localization?

File: js/ui/html_editor/modules/toolbar.js [hunk 4]
```diff
@@ -540,7 +236,7 @@ if(Quill) {
                     hint: localize(buttonText),
                     text: localize(buttonText),
                     icon: iconName.toLowerCase(),
-                    onClick: this._formatHandlers[name] || this._getDefaultClickHandler(name),
+                    onClick: this._formatHandlers[name] || getDefaultClickHandler(name, this),
```
Comment (DokaRus) [inline js/ui/html_editor/modules/toolbar.js]:
```suggestion
                    onClick: this._formatHandlers[name] || getDefaultClickHandler(this, name),
```
`context` then `handler name`
like [Function.prototype.apply()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply) `.apply(context, arg1, arg2, ...)`

Check other exports from toolbar_helper, pls. They're should look consistent with the others

File: js/ui/html_editor/ui/formDialog.js [hunk 5]
```diff
@@ -58,6 +63,7 @@ class FormDialog {
             deferRendering: false,
             focusStateEnabled: false,
             showCloseButton: false,
+            fullScreen: this._getFullScreen(),
```
Comment (DokaRus) [inline js/ui/html_editor/ui/formDialog.js:66]: 👍🏻 
//T1026801

File: js/docEnums.js [hunk 6]
```diff
@@ -1019,6 +1019,11 @@
  * @enum {'background'|'bold'|'color'|'font'|'italic'|'link'|'size'|'strike'|'script'|'underline'|'blockquote'|'header'|'indent'|'list'|'align'|'code-block'}
  */
 
+/**
+ * @typedef {string} Enums.HtmlEditorContextMenuItem
+ * @enum {'background'|'bold'|'color'|'font'|'italic'|'link'|'image'|'strike'|'subscript'|'superscript'|'underline'|'blockquote'|'increaseIndent'|'decreaseIndent'|'orderedList'|'bulletList'|'alignLeft'|'alignCenter'|'alignRight'|'alignJustify'|'codeBlock'|'variable'|'undo'|'redo'|'clear'|'insertTable'|'insertHeaderRow'|'insertRowAbove'|'insertRowBelow'|'insertColumnLeft'|'insertColumnRight'|'deleteColumn'|'deleteRow'|'deleteTable'|'cellProperties|'tableProperties'}
```
Comment (pavelgruba) [inline js/docEnums.js]:
```suggestion
 * @enum {'background'|'bold'|'color'|'font'|'italic'|'link'|'image'|'strike'|'subscript'|'superscript'|'underline'|'blockquote'|'increaseIndent'|'decreaseIndent'|'orderedList'|'bulletList'|'alignLeft'|'alignCenter'|'alignRight'|'alignJustify'|'codeBlock'|'variable'|'undo'|'redo'|'clear'|'insertTable'|'insertHeaderRow'|'insertRowAbove'|'insertRowBelow'|'insertColumnLeft'|'insertColumnRight'|'deleteColumn'|'deleteRow'|'deleteTable'|'cellProperties'|'tableProperties'}
```

**Review by DokaRus — DISMISSED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Review by pavelgruba — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Pull request merged**